lua_cjson.c

```diff
@@ -713,21 +714,37 @@ static void json_append_data(lua_State *l, json_config_t *cfg,
     case LUA_TTABLE:
         current_depth++;
         json_check_encode_depth(l, cfg, current_depth, json);
-        len = lua_array_length(l, cfg, json);
-        if (len > 0 || (len == 0 && !cfg->encode_empty_table_as_object))
+
+        int as_array = 0;
```

Comment:
I'm seeing the following warning on some operating system (maybe CentOS 6?):

```
lua_cjson.c: In function ‘json_append_data’:
lua_cjson.c:731:9: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
         int as_array = 0;
         ^
```

Better always put such auto variable declarations on the beginning of the belonging code block.

I'll fix it on my side.

Reply:
Oh, sorry about that - forgot that lua-cjson would also be compiled with older platforms as well, but definitely an oversight on my part. Thanks!